### ISSUE TITLE
Error in x[i] <- sprintf("<a href=\"%s#%s\"", filenames[which.max(lines[lines <=  :    replacement has length zero Calls: <Anonymous> ... <Anonymous> -> split_chapters -> restore_links -> lapply -> FUN

### ISSUE THREAD

**Issue by GabriellaS-K** [rstudio/bookdown#1044]
Hi, 

I keep getting the same error when I try to build the book: 
```
/Applications/RStudio.app/Contents/MacOS/pandoc/pandoc +RTS -K512m -RTS _main.utf8.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output _main.html --lua-filter /Library/Frameworks/R.framework/Versions/4.0/Resources/library/bookdown/rmarkdown/lua/custom-environment.lua --lua-filter /Library/Frameworks/R.framework/Versions/4.0/Resources/library/rmarkdown/rmarkdown/lua/pagebreak.lua --lua-filter /Library/Frameworks/R.framework/Versions/4.0/Resources/library/rmarkdown/rmarkdown/lua/latex-div.lua --wrap preserve --standalone --section-divs --table-of-contents --toc-depth 3 --template /Library/Frameworks/R.framework/Versions/4.0/Resources/library/bookdown/templates/gitbook.html --highlight-style tango --number-sections --include-in-header /var/folders/bf/s948qjx51kb_vxsf6f4kjc7r0000gn/T//Rtmpe5OBx6/rmarkdown-str3b272532387.html --mathjax --citeproc 
Error in x[i] <- sprintf("<a href=\"%s#%s\"", filenames[which.max(lines[lines <=  : 
  replacement has length zero
Calls: <Anonymous> ... <Anonymous> -> split_chapters -> restore_links -> lapply -> FUN
Please delete _main.Rmd after you finish debugging the error.
Execution halted

Exited with status 1.
```
Here is my session info:

```

> xfun::session_info('rmarkdown')
R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7, RStudio 1.4.1087

Locale: en_GB.UTF-8 / en_GB.UTF-8 / en_GB.UTF-8 / C / en_GB.UTF-8 / en_GB.UTF-8

Package version:
  base64enc_0.1.3 digest_0.6.27   evaluate_0.14   glue_1.4.2     
  graphics_4.0.3  grDevices_4.0.3 highr_0.8       htmltools_0.5.0
  jsonlite_1.7.2  knitr_1.30      magrittr_2.0.1  markdown_1.1   
  methods_4.0.3   mime_0.9        rlang_0.4.9     rmarkdown_2.5.3
  stats_4.0.3     stringi_1.5.3   stringr_1.4.0   tinytex_0.27   
  tools_4.0.3     utils_4.0.3     xfun_0.19       yaml_2.2.1     

Pandoc version: 2.11.2
> 


```

Then I ran:

```
update.packages(ask = FALSE, checkBuilt = TRUE)
```

I would be grateful for any help. I am a relative beginner so please bare this in mind!
---

By filing an issue to this repo, I promise that

- [X] I have fully read the issue guide at https://yihui.org/issue/.
- [X] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('bookdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/bookdown')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [X] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by cderv** [comment]
Which book are you trying to compile ? 
What is the content of the file ? 

There was a similar error in the community where the issue was that each chapter must start with header 1 header (`# chapter`) - for reference https://community.rstudio.com/t/bookdown-debugging-fixing-incomprehensible-error/87014/6

Could you have the same issue ?

**Comment by cderv** [comment]
This is the kind of error message we need to improve also - tracked on https://github.com/rstudio/bookdown/issues/1017

**Comment by GabriellaS-K** [comment]
Hi!

Sorry, what do you mean by which book, and content? 
I've checked, each chapter definitely starts with header 1 header (#), I saw that other error you linked to-tried all that before posting :) 

Thanks for your time

**Comment by cderv** [comment]
> when I try to build the book:

Which book are you trying to build ?

**Comment by GabriellaS-K** [comment]
Ah sorry, I understand
when I try to build the bookdown book, by clicking on 'build book' in the project file. 

Yes, index.Rmd is in the project file. Its weird because I followed the demo, it worked, and then in the evening it didn't.

the YAML in my index reads: 

`---

#####################
## metadata ##
#####################
title             : "Results from Study "

site: bookdown::bookdown_site
documentclass: book
output:
  bookdown::gitbook: 
      highlight: tango
  #bookdown::pdf_book: default
date: "`r Sys.Date()`"


#######################
## bibliography path ##
#######################
bibliography: GK_References.bib

#
---`

Then the next thing in my index: 


# index 
```{r}
options(repos="https://cran.rstudio.com" )

remotes::install_github('rstudio/bookdown')
```
The files in my project are: 

- .Rhistory
- 01_introductions.Rmd
-03_results.Rmd
-99_references.Rmd
-GK_References.bib
-index.Rmd
-main.Rmd

Does this help? I'm sure there is more I can provide you with to understand the problem, I'm really sorry for being slow-I'm still learning.

**Comment by cderv** [comment]
It would be helpful if you follow advice to format correctly 
https://yihui.org/issue/#please-format-your-issue-correctly

>  Its weird because I followed the demo, it worked, and then in the evening it didn't.

Is the bookdown-demo project working when you download it and build the book ? 

> remotes::install_github('rstudio/bookdown')

It is not recommended to install **bookdown** from within a **bookdown** project. You already need **bookdown** to render the project so don't try to install it from within one of the Rmd. 

>  I'm sure there is more I can provide you with to understand the problem

You can share a zip of the project with your files in it and include it in your next answer here as advised in the issue guide https://yihui.org/issue/#bug-reports "If your example contains multiple files, please upload them as a zip file to the Github issue."

Thank you

**Comment by GabriellaS-K** [comment]
Hi, 

Thanks so much. I ended up watching a [video tutorial ](https://t.co/nzonK5uEMy?amp=1) that discusses bookdown, and I found this really helpful. I went back to start over, made a new clean project, and am adding files one by one. This seems to 'build' properly. 

> It is not recommended to install bookdown from within a bookdown project. You already need bookdown to render the project so don't try to install it from within one of the Rmd.
Ah, OK that's good to know, I didn't realise. So the best is to open R, and install bookdown, then open the project I'm working on? Do I need to do this every time I use bookdown?

Thanks again for your help
